pkg/sessionctx/variable/tidb_vars.go

```diff
@@ -966,6 +966,10 @@ const (
 	// DivPrecisionIncrement indicates the number of digits by which to increase the scale of the result of
 	// division operations performed with the / operator.
 	DivPrecisionIncrement = "div_precision_increment"
+
+	// TiDBEnableSharedLockPromotion indicates whether the `select for share` statement would be executed
+	// as `select for update` statements which do acquire pessimistic locks.
+	TiDBEnableSharedLockPromotion = "tidb_enable_shared_lock_promotion"
```

Comment:
Should we add this new variable into the Plan Cache Key? Otherwise the optimizer might reuse an old plan after switching this variable, which might cause some problem.